src/Compilers/Core/Portable/CodeGen/ITokenDeferral.cs

```diff
@@ -13,7 +13,12 @@ internal interface ITokenDeferral
         uint GetFakeSymbolTokenForIL(Cci.ISignature value, SyntaxNode syntaxNode, DiagnosticBag diagnostics);
         uint GetSourceDocumentIndexForIL(Cci.DebugSourceDocument document);
 
-        Cci.IFieldReference GetFieldForData(ImmutableArray<byte> data, SyntaxNode syntaxNode, DiagnosticBag diagnostics);
+        Cci.IFieldReference GetFieldForData(ImmutableArray<byte> data, ushort alignment, SyntaxNode syntaxNode, DiagnosticBag diagnostics);
+
+        /// <summary>Gets a field that may be used to lazily cache an array created to store the specified data.</summary>
+        /// <remarks>This is used to cache an array created with the exact same arguments passed to <see cref="GetFieldForData"/>.</remarks>
+        Cci.IFieldReference GetArrayCachingFieldForData(ImmutableArray<byte> data, ushort alignment, Cci.IArrayTypeReference arrayType, SyntaxNode syntaxNode, DiagnosticBag diagnostics);
```

Comment:
>ushort alignment [](http://example.com/codeflow?start=83&length=16)

This `alignment` parameter is confusing. Do we really need it? 
The only call-site currently passes the element size, which doesn't seem relevant when creating a field of array type. 
Consider removing it and hard-coding an `alignment: 1` where this value is eventually used (call to `GenerateDataFieldName` in `CreateArrayCachingField`).